### PR TITLE
FAST.Farm-C++ API Bug Fixes

### PR DESCRIPTION
**Pull request by rafmudaf**
**Feature or improvement description**
The C++ API tests fail in the automated system in the FAST.Farm branch. This pull request will address those problems.

### CODE REVIEW

**Comment by jjonkman** [comment]
@rafmudaf -- It appears the cpp-interface-tests are still failing :-(.

**Comment by rafmudaf** [comment]
> @rafmudaf -- It appears the cpp-interface-tests are still failing :-(.

Yes but different error now. Even worse - seg fault.

**Comment by rafmudaf** [comment]
Screen Shot 2021-04-02 at 4 10 34 PM

Here's where the debugger stops. @jjonkman @bjonkman does this give you any hint?

**Comment by rafmudaf** [comment]
@gantech does this seg fault hint at a particular problem?
You can see the changes to the C++ api and OpenFAST library in https://github.com/OpenFAST/openfast/pull/584.

**Comment by rafmudaf** [comment]
@jjonkman I trust you've heard the good news. Just waiting on the tests in GH Actions.

**Comment by jjonkman** [comment]
@rafmudaf -- Yes; I assume you want me to merge this in as soon as all tests pass (crossing fingers)?

**Comment by rafmudaf** [comment]
@gantech could you please review the changes here?

File: glue-codes/openfast-cpp/src/OpenFAST.cpp [hunk 1]
```diff
@@ -822,8 +832,10 @@ void fast::OpenFAST::allocateMemory() {
     cDriver_Input_from_FAST.resize(nTurbinesProc) ;
     cDriver_Output_to_FAST.resize(nTurbinesProc) ;
     
-  if(!scStatus)
-      scio.from_SC.resize(nTurbinesProc);
+    if(scStatus) {
+        std::cout << "Use of Supercontroller is not supported through the C++ API right now" << std::endl;
```
Comment (bjonkman) [inline glue-codes/openfast-cpp/src/OpenFAST.cpp:836]: Couldn't this whole if statement be removed anyway? This gets resized in the supercontroller::init, too.

**Pull request merged**